### PR TITLE
Add shellcheck to lint shell scripts

### PR DESCRIPTION
**Pull request by mlangbehn**
Now, all shell scripts are linted for best practices.

Requires: #64, #66, #70, and #71
Signed-off-by: Matt Langbehn <matthew.langbehn@gmail.com>

### CODE REVIEW

File: roles/zuul/files/jobs/hoist.yaml
Comment (eggshell) [inline roles/zuul/files/jobs/hoist.yaml:25]: This is a nit to the nth degree, but can we keep tests of the same file extension grouped together like they are in `.travis.yml`?

**Review by eggshell — CHANGES_REQUESTED** [review]
One nit inline. Other than that we just need #66 to merge first.

File: roles/zuul/files/jobs/hoist.yaml
Comment (mlangbehn) [reply roles/zuul/files/jobs/hoist.yaml:25]: We certainly can. I added this test to the end because I'm not entirely sure how this mechanism works. In Travis CI, we are running the tests in parallel, but I think they happen sequentially here. If they are sequential, I think it makes sense to add this check to the end?

Comment (eggshell) [reply roles/zuul/files/jobs/hoist.yaml:25]: They happen in parallel iirc.

Similar to: https://blog.travis-ci.com/2012-11-28-speeding-up-your-tests-by-parallelizing-them/

Comment (mlangbehn) [reply roles/zuul/files/jobs/hoist.yaml:25]: If they were happening in parallel, wouldn't each line end with an `&`?

Comment (eggshell) [reply roles/zuul/files/jobs/hoist.yaml:25]: I'm not sure why that would be needed since it seems to take each line as a separate thing to run and it goes off and runs each one (all starting at the same time).

Comment (mlangbehn) [reply roles/zuul/files/jobs/hoist.yaml:25]: Just rearranged all three to be in alphabetical order.

**Review by eggshell — APPROVED** [review]
lgtm once #66 merges and this passes CI fully.

**Comment by rattboi** [comment]
recheck

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/67/1485194029.72/ubuntu-hoist : FAILURE in 23s
- http://logs.bonnyci.com/check_github/BonnyCI/hoist/67/1485194029.72/ansible-syntax : SUCCESS in 53s
- http://logs.bonnyci.com/check_github/BonnyCI/hoist/67/1485194029.72/hoist-checks : SUCCESS in 8s

**Comment by mlangbehn** [comment]
recheck

**Pull request merged**
